src/docdb/nodes.ts

```diff
@@ -123,6 +123,10 @@ export class DocDBCollectionNode implements INode {
 	addNewDocToCache(document: any): void {
 		this._children.unshift(new DocDBDocumentNode(document.id, this, document))
 	}
+
+	removeNodeFromCache(documentNode: DocDBDocumentNode): void {
+		this._children = this._children.filter(doc => doc.id !== documentNode.id);
```

Comment:
This should work, but the normal pattern is to use the 'splice' function to remove from an array:
```
const index: number = this._children.findIndex(doc => doc.id !== documentNode.id);    
if (index !== -1) {
    array.splice(index, 1);
}
```